### ISSUE TITLE
[BUG]final result = await PhotoManager.editor.deleteWithIds(ids)，接受不到返回值。

### ISSUE THREAD

**Issue by J999999** [CaiJingLong/flutter_photo_manager#362]
在android环境下，删除本地相册图片， 图片删除成功，但是
final result = await PhotoManager.editor.deleteWithIds(ids);
    print('---------- delete, ${result.isNotEmpty}');
在result后的代码不执行。（也就是print这行代码不执行， 跑的demo也是同样情况）。是flutter版本问题还是其他，望解答。 急急急。

**Comment by CaiJingLong** [comment]
android os version?

**Comment by leonardocustodio** [comment]
I'm having the same problem. Already tried a lot of different async code but it never works. The file is deleted but the code stops running. Here is an example:

   ```
 print('Before photo manager delete: ${entity.id}');
    await PhotoManager.editor.deleteWithIds([entity.id]).then((value) {
      print('Photo Editor Result: ${value}');
    }).catchError((error) {
      print(error);
    });
```

The block inside doesn't run. But again, the picture is deleted. Here is my flutter doctor output:

`[√] Flutter (Channel beta, 1.21.0-9.2.pre, on Microsoft Windows [Version 10.0.19041.450], locale pt-BR)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[√] Android Studio (version 4.0)
`

Also I'm running this emulator:

```
Name: Pixel_3a_API_29
CPU/ABI: Google Play Intel Atom (x86)
hw.device.name: pixel_3a
image.androidVersion.api: 29
tag.id: google_apis_playstore
hw.camera.front: emulated
avd.ini.displayname: Pixel 3a API 29
image.sysdir.1: system-images\android-29\google_apis_playstore\x86\
tag.display: Google Play
```

Regards,
Leonardo

**Comment by leonardocustodio** [comment]
By the way this also happens on iOS. It is not a android only issue.

**Comment by leonardocustodio** [comment]
I just saw this PR #354 does this fixes this issue? Anyone has test it out?

**Comment by leonardocustodio** [comment]
I've just tested it using:

```
  photo_manager:
    git:
      url: https://github.com/CaiJingLong/flutter_photo_manager.git
      ref: master
```

And I can confirm that the problem is still happening. Using the master branch the pic is not even deleted. On version 0.5.8 the pic is at least deleted the problem is that the code doesn't keep running.

**Comment by SUNbrightness** [comment]
> 



> android os version?

android version:10

**Comment by J999999** [comment]
> android os version?

10,    
_clickDeleteAction() {
    PhotoManager.editor.deleteWithIds([widget.id]);
    Navigator.pop(context);
  } 
我目前是这么使用的， 但是不能在删除操作执行完以后返回， 很难受。

**Comment by leonardocustodio** [comment]
Yep without the await it works but we won't know for sure if the picture was really deleted.

Still I found that when compiling using sdkVersion 29 in my device at least it only worked when combining with the permission:

`android:requestLegacyExternalStorage="true"`

On the AndroidManifest.

**Comment by CaiJingLong** [comment]
Fixed in #367.
Please wait for 0.6.0 version.
